marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs

```diff
@@ -546,6 +546,22 @@ class FlowUtils {
     if(options.file) {
         headers["createdUsingFile"] = options.file;
     }
+
+    const sourceName = options.sourceName ? options.sourceName : "";
+    const sourceType = options.sourceType ? options.sourceType : "";
+
+    if(headers["sources"] && !Array.isArray(headers["sources"])) {
+      let tempSourcesArray = [];
+      tempSourcesArray.push(headers["sources"]);
+      headers["sources"] = tempSourcesArray;
+    }
+
+    if(sourceName || sourceType) {
+      if(!headers["sources"] || headers["sources"].length == 0) {
+        headers["sources"] = [];
+      }
+      headers["sources"].push({"name": sourceName, "datahubSourceType": sourceType});
```

Comment:
Let's only populate something if it's not null - this will result in "datahubSourceType": "" if it's not specified. So above, default sourceName and sourceType to null if they're not in options. And then only do any of this code if either sourceName or sourceType is not null. And only add them to the source object if they're not null.